doctrine.rst

```diff
@@ -632,7 +676,212 @@ Now, simplify your controller::
+
+    // src/Controller/ProductController.php
+    namespace App\Controller;
+
+    use App\Entity\Product;
+    use App\Repository\ProductRepository;
+    use Symfony\Bridge\Doctrine\Attribute\MapEntity;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\Routing\Annotation\Route;
+    // ...
+
+    class ProductController extends AbstractController
+    {
+        #[Route('/product/{id}')]
+        public function show(
+            #[MapEntity]
+            Product $product
+        ): Response {
+            // use the Product!
+            // ...
+        }
+    }
+
+Fetch Automatically
+~~~~~~~~~~~~~~~~~~~
+
+If your route wildcards match properties on your entity, then the resolver
+will automatically fetch them:
+
+.. configuration-block::
+
+    .. code-block:: php-attributes
+
+        /**
+         * Fetch via primary key because {id} is in the route.
+         */
+        #[Route('/product/{id}')]
+        public function showByPk(Post $post): Response
+        {
+        }
+
+        /**
+         * Perform a findOneBy() where the slug property matches {slug}.
+         */
+        #[Route('/product/{slug}')]
+        public function showBySlug(Post $post): Response
+        {
+        }
+
+Automatic fetching works in these situations:
+
+* If ``{id}`` is in your route, then this is used to fetch by
+  primary key via the ``find()`` method.
+
+* The resolver will attempt to do a ``findOneBy()`` fetch by using
+  *all* of the wildcards in your route that are actually properties
+  on your entity (non-properties are ignored).
+
+You can control this behavior by actually *adding* the ``MapEntity``
+attribute and using the `MapEntity options`_.
+
+Fetch via an Expression
+~~~~~~~~~~~~~~~~~~~~~~~
+
+If automatic fetching doesn't work, use an expression:
+
+.. configuration-block::
+
+    .. code-block:: php-attributes
+
+        use Sensio\Bundle\FrameworkExtraBundle\Configuration\Entity;
+
+        #[Route('/product/{product_id}')]
+        public function show(
+            #[MapEntity(expr: 'repository.find(product_id)')]
+            Product $product
+        ): Response {
+        }
+
+Use the special ``MapEntity`` attribute with an ``expr`` option to
+fetch the object by calling a method on your repository. The
+``repository`` method will be your entity's Repository class and
+any route wildcards - like ``{product_id}`` are available as variables.
+
+This can also be used to help resolve multiple arguments:
+
+.. configuration-block::
+
+    .. code-block:: php-attributes
+
+        #[Route('/product/{id}/comments/{comment_id}')]
+        public function show(
+            Product $product
+            #[MapEntity(expr: 'repository.find(comment_id)')]
+            Comment $comment
+        ): Response {
+        }
+
+In the example above, the ``$product`` argument is handled automatically, 
+but ``$comment`` is configured with the attribute since they cannot both follow
+the default convention.
```

Comment:
A note should be added to explain that `symfony/expression-language`  is require to be able to use the `expr` option